main.go

```diff
@@ -324,3 +353,238 @@ func printList() {
 	}
 	fmt.Println(b.String())
 }
+
+// locateFile locates a file named file, or a directory if name is empty, and returns it's full path
+// It first tries in the paths given by the dirs, and then a system lookup
+func locateFile(file string, dirs []string) string {
+	var res string
+	for _, dir := range dirs {
+		if dir != "" {
+			fi, err := os.Stat(filepath.Join(dir, file))
+			if err == nil && ((file == "" && fi.IsDir()) || (file != "" && fi.Mode().IsRegular())) {
+				return filepath.Join(dir, file)
+			}
+		}
+	}
+	if file != "" && res == "" {
+		p, _ := exec.LookPath(file)
+		if p != "" {
+			return p
+		}
+	}
+	return ""
+}
+
+// getBPFObject finds or builds ebpf object file and returns it's path
+func getBPFObject() (string, error) {
+	bpfObjFileName := "tracee.bpf.o"
+	exePath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+	//locations to search for the bpf file, in the following order
+	searchPaths := []string{
+		os.Getenv("TRACEE_BPF_FILE"),
+		filepath.Dir(exePath),
+		traceeInstallPath,
+	}
+	bpfObjFilePath := locateFile(bpfObjFileName, searchPaths)
+	if bpfObjFilePath != "" && debug {
+		fmt.Printf("found bpf object file at: %s\n", bpfObjFilePath)
+	}
+	if bpfObjFilePath == "" {
+		if debug {
+			fmt.Printf("could not find tracee bpf object file in any of %v. attempting to build it\n", searchPaths)
+		}
+		madeBPFObjFilePath := filepath.Join(traceeInstallPath, bpfObjFileName)
+		err = makeBPFObject(madeBPFObjFilePath)
+		if err != nil {
+			return "", err
+		}
+		if debug {
+			fmt.Printf("succesfully built ebpf obj file into: %s\n", madeBPFObjFilePath)
+		}
+		bpfObjFilePath = madeBPFObjFilePath
+	}
+
+	return bpfObjFilePath, nil
```

Comment:
What if we want to force re-comilation of the bpf code?
For example, when new kernel is installed?
Maybe we shuold fallback to compilation if bpf code loading failed?

Reply:
I plan to add "install" and "uninstall" commands to tracee. for now the user can just remove /opt/tracee and it will recompile